### PR TITLE
Add UPGRADING.md to document breaking changes

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Add UPGRADING.md to document breaking changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Upgrade Guidance
+## 1.x.y to 2.0.0
+
```
Comment (dblock) [inline UPGRADING.md:3]: Remove double space.

File: UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# Upgrade Guidance
```
Comment (dblock) [inline UPGRADING.md:1]: I would say "Upgrading OpenSearch.Net" here.

Add an empty line after ;)

**Review by dblock — CHANGES_REQUESTED** [review]
Add a link to UPGRADING in README.md. In [other projects](https://github.com/ruby-grape/grape?tab=readme-ov-file#stable-release) I've seen things like "You are reading the documentation for the next release of X. See [UPGRADING](UPGRADING.md) for upgrading from a previous version.", but an entry in the list of the links would suffice too.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
tests are failing ...

**Comment by Xtansia** [comment]
> tests are failing ...

@dblock Failing tests are unrelated one is a known flaky test (#443) others are breakages from building the unreleased k-nn plugin

**Pull request merged**